backend/routes/users

```diff
@@ -3,25 +3,50 @@ var router = express.Router();
 const connection = require('../conn');
 const mysql = require('mysql2');
 
+
 router.post('/', (req, res) => {
-    let newUser = req.body;
-    console.log(req.body);
+  let newName = req.body.newName;
+  connection.connect(function (err) {
+    if (err) {
+        console.log(err);
+        req.sendStatus(500);
+        return
+    }
+    let sql = `SELECT userName FROM users WHERE userName = ${mysql.escape(newName)}`         
+    connection.query(sql, function(err, result) {
+      if (err) {
+        console.log(err)
+        res.send(404)
+        return
+      }
+      if (result.length == 0) {
+        saveUser(req, res);
+      } else {
+        res.sendStatus(404)
+      }
+    })
+  })
+}) 
+
 
+function saveUser (req, res) {
+    let userName = req.body.newName;
     connection.connect(function (err) {
```

Comment:
Vi behöver ( och bör inte ) skriva connect metoden mer än en gång. Som jag förstått det. Testa att helt enkelt bort denna del.

Reply:
Det förklarade varför det tog tid att logga in. Ändrat.